src/d3a/models/area/event_dispatcher.py

```diff
@@ -236,6 +263,32 @@ def _get_agents_for_market_type(
             return dispatcher_object.settlement_agents
         assert False, f"Market type not supported {market_type}"
 
+    @property
+    def _should_agent_be_created(self) -> bool:
+        if not self.area.parent:
+            return False
+        if self.area.strategy:
+            return False
+        if not self.area.parent.events.is_connected:
+            return False
+        if not self.area.children:
+            return False
+        return True
+
+    def create_area_agents_for_future_markets(self, market: Market) -> None:
```

Comment:
I think this method is not needed and can be incorporated in the create_area_agents method. I will try to tackle it in the context of feature/D3ASIM-3633

Reply:
Hmm, I was thinking about this a lot, too. But then you would need a check for the market_type in `create_area_agents` which dies not make it more pretty. This method is already called from a future market specific method (`area.markets.activate_future_markets`). So, IMO it is good like it is here.